sdks/python/apache_beam/runners/common.py

```diff
@@ -1941,3 +1945,64 @@ def validate_transform(transform_id):
 
   for t in pipeline_proto.root_transform_ids:
     validate_transform(t)
+
+
+def merge_common_environments(pipeline_proto):
```

Comment:
Thanks! It would be good to have a unit test to verify it works as expected

Reply:
Added a couple of tests as well as verifying manually on more complex pipelines.